src/Microsoft.Data/BaseColumn.BinaryOperators.cs

```diff
@@ -371,67 +311,6 @@ public abstract partial class BaseColumn
         {
             return column.And(value);
         }
-
-        public static BaseColumn operator &(BaseColumn column, byte value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, char value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, decimal value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, double value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, float value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, int value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, long value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, sbyte value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, short value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, uint value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, ulong value)
-        {
-            return column.And(value);
-        }
-
-        public static BaseColumn operator &(BaseColumn column, ushort value)
-        {
-            return column.And(value);
-        }
-
         public static BaseColumn operator |(BaseColumn left, BaseColumn right)
```

Comment:
Maybe this should change to take `PrimitiveColumn<bool>` as its parameters and also return a `PrimitiveColumn<bool>`?

Reply:
Unfortunately no, because we can't specialize this template. If I changed all the `BaseColumn` here to `PrimitiveColumn<bool>`, the compiler will complain that the overload should contain at least one defining type. Also, `PrimitiveColumn<T>` is a different type than `PrimitiveColumn<bool>`, so I can't overload this in `PrimitiveColumn.cs` either. I don't think I can define an extension method that accepts a `PrimitiveColumn<bool>` either, but I'm not certain yet.